src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
```

Comment:
Ah, you are saying if a base produces gets overriden by the derived action, then both of them set the value but the one closer to the action wins? ok.. that makes sense.